### PR TITLE
fix: proxying request query search (fixes #133)

### LINKED ISSUE #133: edge gateway does not proxy query parameters to some of the resolution layers
https://github.com/web3-storage/reads/blob/main/packages/edge-gateway/src/gateway.js#L102

IPFS Gateway might have request parameters, like `format=car`. In  `1st layer resolution - CDN` we just proxy the request object, and consequently query parameters will be sent. However, other resolution layers are lacking passing the query parameters.

[getFromDotstorage](https://github.com/web3-storage/reads/blob/main/packages/edge-gateway/src/gateway.js#L102) and [getFromGatewayRacer](https://github.com/web3-storage/reads/blob/main/packages/edge-gateway/src/gateway.js#L119) only pass CID, pathname and headers, which means that for an URL like 

```js
const urlString = 'https://bafy.ipfs.w3slink.link/path/to/file?format=car'
const url = new URL(urlString)
```

pathname will include `path/to/file`, but we drop query parameters that will result in not getting appropriate response from other parties. We need to proxy also search params from URL

### PR DESCRIPTION
**Pull request by jsdevel**

### CODE REVIEW

File: packages/edge-gateway/test/index.spec.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ test('Gets content', async (t) => {
   const { mf } = t.context
 
   const response = await mf.dispatchFetch(
-    'https://bafkreihl44bu5rqxctfvl3ahcln7gnjgmjqi7v5wfwojqwriqnq7wo4n7u.ipfs.localhost:8787'
+    'https://bafkreihl44bu5rqxctfvl3ahcln7gnjgmjqi7v5wfwojqwriqnq7wo4n7u.ipfs.localhost:8787?foo=test'
```
Comment (vasco-santos) [inline packages/edge-gateway/test/index.spec.js:55]: would be cool to add a test here specifically for search params. Then adding a mock in smoke https://github.com/web3-storage/reads/tree/main/packages/edge-gateway/test/mocks/ipfs.io to handle a different response there :)

**Review by vasco-santos — COMMENTED** [review]
Everything is looking good, but needs a test

File: packages/edge-gateway/test/index.spec.js (hunk 1 above)
Comment (e-schneid) [reply packages/edge-gateway/test/index.spec.js]: @vasco-santos I've added a new test here.  I'm unfamiliar with the codebase, so I'm still not exactly sure why the fixtures being added to the container seem to take precedence over the mocks managed by smoke (i.e., the `fetch` will return the fixture if it finds one and the smoke function never gets invoked) but it seemed to finally work by adding a new cid only to the smoke file (not the fixture).  🤷 

Let me know if there's anything you'd like changed.

File: packages/ipfs-gateway-race/lib/index.js [hunk 2]
```diff
@@ -135,14 +137,15 @@ export async function gatewayFetch (
   gwUrl,
   cid,
   pathname,
+  search,
```
Comment (vasco-santos) [inline packages/ipfs-gateway-race/lib/index.js:140]: Should we put the search as optional within options object?

Thinking that this change would mean a breaking change, as well as search params not being quite common. So, being forced to add a `''` string as param here seems weird.

**Review by vasco-santos — CHANGES_REQUESTED** [review]
Looking good the tests overall. We just need to make the mock in a different file. Also, just found a minor change that we should do in the gateway race

File: packages/edge-gateway/test/index.spec.js (hunk 1 above)
Comment (vasco-santos) [reply packages/edge-gateway/test/index.spec.js]:
```
so I'm still not exactly sure why the fixtures being added to the container seem to take precedence over the mocks managed by smoke
```

The tests are setup as a request race of between a IPFS node running inside a docker container, and two other mock servers. Mock serves in this case should be faster and return asap. So, I guess in this case we need to add a new mock file with a different more specific name. See https://github.com/sinedied/smoke#query-parameters

File: packages/ipfs-gateway-race/lib/index.js (hunk 2 above)
Comment (e-schneid) [reply packages/ipfs-gateway-race/lib/index.js]: makes sense to me.  updated

File: packages/edge-gateway/test/index.spec.js (hunk 1 above)
Comment (e-schneid) [reply packages/edge-gateway/test/index.spec.js]: Updated.  I've removed the else blocks in the mocks with the query param checks and added the new more specific query param mock files.

**Pull request merged**